Fix loading of negative ROI coordinates

scottclowe opened the pull request:
The ROI coordinates are stored as signed (not unsigned) integers,
but the `imagej.read_roi` function loading method was only valid
for unsigned integers.

This was causing a problem specifically when ROIs were drawn
outside of the image (which is possible in ImageJ by dragging a
box outside the window edge). In this case, the top or left coordinate
values would be returned as very large positive values.

This problem is fixed, but I don't know which other values in the
ImageJ file are stored as signed as opposed to unsigned integers, so
other values may still be loaded incorrectly.

pkaifosh commented:
Thanks @scottclowe and @swkeemink. 

There are a couple of flake8 Once you fix the flake8 issues that need to be fixed before we can merge so that the travis-ci checks pass:
* sima/misc/imagej.py:102:80: E501 line too long (80 > 79 characters)
* sima/misc/imagej.py:108:14: E225 missing whitespace around operator

Also, do negative ROI coordinates cause any problems in the downstream processing? E.g. have you tried extracting ROIs with negative coordinates that have been imported in this way?

Finally, please feel free to add yourselves to the testing/bug fixes section of the CREDITS file.

scottclowe commented:
I fixed the flake8 errors, and rebased the fix into my previous commit.
Also I updated the CREDITS file, thanks!

We don't actually use many of SIMA's features in our workflow, so I can't say how negative ROI coordinates would impact on the rest of SIMA. After fixing this issue, we ended up having to resize our "offscreen" ROIs so they were "onscreen" in any case for our own processing pipeline to work.

It might be worthwhile to consider how you want to handle offscreen ROIs generally, presumably by either raising an Error or a Warning since users will probably want to be alerted to them instead of them being silently allowed, and adding appropriate unit tests?

pkaifosh commented:
@scottclowe I think your suggestion of issuing a warning when ROIs are out of bounds is a good one. Please feel free to add this. Also, if you can run any of the downstream analysis just to confirm it still works, that would be great to check before merging. Ideally you could add a test case for extraction with negative ROI coordinates. Two cases come to mind:
* All coordinates negative, and the ROI never in bounds
* Some coordinates negative, with the ROI partly in bounds


@nbdanielson @jzaremba Does ROI registration / transformation between images result in ROIs that are out of bounds or have negative coordinates? Have these extracted fine without problems? Do you think it's safe to merge this as is?

jzaremba commented:
Closing for now. Please re-open if this would still be useful and we can look at it again.
